pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -495,65 +499,172 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
+			} else {
+				log.Log.Infof("The network interface type was changed from virtio to e1000 due to unsupported interface type by qemu slirp network")
+			}
 
-		domain.Spec.Devices.Interfaces = make([]Interface, 0)
-		for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-			if _, ok := networks[iface.Name]; !ok {
-				return fmt.Errorf("Fail to match the Name of a Network")
+			domainIface := Interface{
+				Model: &Model{
+					Type: slirpInterfaceType,
+				},
+				Type: "user",
+				Alias: &Alias{
+					Name: iface.Name,
+				},
 			}
-			if iface.Bridge != nil {
-				domain.Spec.Devices.Interfaces = append(domain.Spec.Devices.Interfaces, Interface{
-					Model: &Model{
-						Type: interfaceType,
-					},
-					Type: "bridge",
-					Source: InterfaceSource{
-						Bridge: DefaultBridgeName,
-					}})
-			} else if iface.Slirp != nil {
-				// TODO: maybe add interface model to vm spec
-				// Slirp configuration works only with e1000 or rtl8139
-				if interfaceType == "virtio" {
-					interfaceType = "e1000"
-				}
-
-				// Proxy is not added as interface, Need to be added as qemu commandlist
-				// Create network interface
-				if domain.Spec.QEMUCmd == nil {
-					domain.Spec.QEMUCmd = &Commandline{}
-				}
-
-				if domain.Spec.QEMUCmd.QEMUArg == nil {
-					domain.Spec.QEMUCmd.QEMUArg = make([]Arg, 0)
-				}
-				domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, Arg{Value: "-device"})
-				domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, Arg{Value: fmt.Sprintf("%s,netdev=%s", interfaceType, iface.Name)})
-				// The network itself will be created on preStartHook
+			domain.Spec.Devices.Interfaces = append(domain.Spec.Devices.Interfaces, domainIface)
+
+			// Create network interface
+			if domain.Spec.QEMUCmd == nil {
+				domain.Spec.QEMUCmd = &Commandline{}
 			}
 
+			if domain.Spec.QEMUCmd.QEMUArg == nil {
+				domain.Spec.QEMUCmd.QEMUArg = make([]Arg, 0)
+			}
+
+			err := createSlirpNetwork(iface, *net, domain)
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	return nil
 }
 
+func createSlirpNetwork(iface v1.Interface, network v1.Network, domain *Domain) error {
+	qemuArg := Arg{Value: fmt.Sprintf("user,id=%s", iface.Name)}
+
+	err := configVMCIDR(&qemuArg, iface, network)
+	if err != nil {
+		return err
+	}
+
+	err = configDNSSearchName(&qemuArg)
+	if err != nil {
+		return err
+	}
+
+	err = configPortForward(&qemuArg, iface)
+	if err != nil {
+		return err
+	}
+
+	domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, Arg{Value: "-netdev"})
+	domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, qemuArg)
+
+	return nil
+}
+
+func configPortForward(qemuArg *Arg, iface v1.Interface) error {
+	if iface.Slirp.Ports == nil {
+		return nil
+	}
+
+	for _, forwardPort := range iface.Slirp.Ports {
+
+		if forwardPort.Port == 0 {
+			return fmt.Errorf("Port must be configured")
+		}
+
+		// Check protocol, its case sensitive like kubernetes
```

Comment:
not sure where this comment fits, since we don't seem to look / care about case of the protocol anywhere here.

Reply:
removed